### PR TITLE
feat(app): use React's Context API to prevent redundant current user query

### PR DESCRIPTION
**Pull request by 24thsaint**
# Enhancements
- Current user querying is now being handled by React's Context API (UserContextProvider). This avoids querying the current user everytime the app navigates between pages resulting in faster page loads.

- Moderator selection is now more emphasized.

# Fixes
- Grammar errors.

- Search result limit from 25 to 24 results per page.

### CODE REVIEW

File: src/contexts/CurrentUserContext.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import React from 'react'
+import Profile from '../models/Profile'
+
+const currentUser = new Profile()
+const CurrentUserContext = React.createContext({
+  currentUser,
+  updateCurrentUser: (profile: Profile) => {
+    // Overwrite implementation
+  },
```
Comment (jglosbanes) [inline src/contexts/CurrentUserContext.ts]: Work in Progress pa ini or ga work na tanan?

File: src/pages/Profile.tsx [hunk 2]
```diff
@@ -68,22 +70,23 @@ class ProfilePage extends Component<CheckoutProps, ProfilePageState> {
...
     }
 
-    let currentUser: Profile
-    if (id) {
-      currentUser = await Profile.retrieve()
-    } else {
-      currentUser = profile
-    }
+    const currentUser: Profile = this.props.currentUser
+    console.log({ currentUser })
```
Comment (jglosbanes) [inline src/pages/Profile.tsx]: Need pa ni?

Comment (jglosbanes) [inline src/pages/Profile.tsx]: need pa ni ang console.log()?

File: src/contexts/CurrentUserContext.ts (hunk 1 above)
Comment (jglosbanes) [inline src/contexts/CurrentUserContext.ts]: Work in progress pa lang ini or ga work na ang tanan?

File: src/pages/Profile.tsx (hunk 2 above)
Comment (jglosbanes) [inline src/pages/Profile.tsx]: Need gid ini ang console.log()?

File: src/models/Search.ts [hunk 3]
```diff
@@ -114,7 +114,7 @@ class Search implements State {
       nextStart: 0,
     },
     paginate: {
-      limit: 25,
+      limit: 24,
```
Comment (jglosbanes) [inline src/models/Search.ts]: Intentional ini or just a forgotten experimental random value nga na change?

File: .gitignore [hunk 4]
```diff
@@ -28,3 +28,5 @@ package-lock.json
 /public/electron.js
 /public/LocalServer.js
 .vscode/launch.json
+
+/cypress/screenshots
```
Comment (jglosbanes) [inline .gitignore]: Ibutang lang ang `/cypress/screenshots ` sa babaw sang `.vscode/launch.json` 
Butang lang dayun sang `EOL`

File: src/contexts/CurrentUserContext.ts (hunk 1 above)
Comment (24thsaint) [reply src/contexts/CurrentUserContext.ts]: This is an abstract method and implementation details will be overwritten in App.tsx.

I'll rewrite the comment to make it more detailed.

File: src/pages/Profile.tsx (hunk 2 above)
Comment (24thsaint) [reply src/pages/Profile.tsx]: Woops, nope, my bad... Will remove it.

File: src/models/Search.ts (hunk 3 above)
Comment (24thsaint) [reply src/models/Search.ts]: We are currently showing `four` results per row,
so limiting at `25` will look weird as there is one uneven item on the last row.

image

This is how it'll look when we limit at `24`.

image

File: .gitignore (hunk 4 above)
Comment (24thsaint) [reply .gitignore]: Got it.

**Pull request merged**